modules/consensus/synchronize.go

```diff
@@ -392,6 +422,11 @@ func (cs *ConsensusSet) rpcRelayHeader(conn modules.PeerConn) error {
 	// If the header is valid and extends the heaviest chain, fetch, accept it,
 	// and broadcast it.
 	go func() {
+		if cs.tg.Add() != nil {
```

Comment:
same here, is there a reason that we want this handling done in a separate thread? It's already in a separate thread and the current thread has nothing more to do.

Does it close the conn with the remote gateway?